### PR TITLE
Improved PreKey refill performance

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: src/main/ts/Cryptobox.ts [hunk 1]
```diff
@@ -29,11 +30,16 @@ export class Cryptobox extends EventEmitter {
       throw new Error(`You cannot initialize Cryptobox without a storage component.`);
     }
 
-    this.logger = new Logdown({prefix: 'cryptobox.Cryptobox', alignOutput: true});
+    if (minimumAmountOfPreKeys > 65535) {
+      minimumAmountOfPreKeys = 65535;
+    }
```
Comment (herrmannplatz) [inline src/main/ts/Cryptobox.ts]: we had a const for the maxium prekey id or?

File: src/main/ts/Cryptobox.ts [hunk 2]
```diff
@@ -123,52 +132,55 @@ export class Cryptobox extends EventEmitter {
       });
   }
 
+  private publish_event(topic: string, event: any): void {
+    this.emit(topic, event);
+    this.logger.log(`Published event "${topic}".`, event);
+  }
```
Comment (herrmannplatz) [inline src/main/ts/Cryptobox.ts]: why wrapping the emit?

File: test/js/specs/store/IndexedDBSpec.js [hunk 3]
```diff
@@ -24,19 +24,97 @@ describe('cryptobox.store.IndexedDB', function() {
...
+          expect(alice.desktop.pk_store.prekeys.length).toBe(0);
+          expect(alice.desktop.publish_prekeys).not.toHaveBeenCalled();
+          return alice.desktop.decrypt('to_bob_desktop', ciphertext);
+        })
+        .then(function(plaintext) {
+          expect(alice.desktop.pk_store.prekeys.length).toBe(0);
+          expect(alice.desktop.cachedSessions.size()).toBe(1);
+          expect(alice.desktop.pk_store.release_prekeys.calls.count()).toEqual(1);
+          expect(alice.desktop.publish_prekeys.calls.count()).toEqual(1); // Published PreKey ID "2"
+          expect(sodium.to_string(plaintext)).toBe(messageFromBob);
```
Comment (herrmannplatz) [inline test/js/specs/store/IndexedDBSpec.js]: why not checking for the actual prekey id? the comment doesnt ensure that its 2 :)

**Comment by bennyn** [comment]
Deleted in favor of https://github.com/wireapp/wire-webapp-cryptobox/pull/69

**Pull request closed without merging**